pkg/git/repository.go

```diff
@@ -55,8 +55,13 @@ func (r *Repository) CheckoutAndCreate(branch string) error {
 	return err
 }
 
-func (r *Repository) Walk(base string, cb func(prefix, name string) error) error {
```

Comment:
@bigkevmcd  wrote,
> I think it might be nicer to have a "Local" implementation of the Source interface.
> All it needs to do is implement Walk, based on the local path, rather than hacking the Repository Walk to do double-duty.

If that avoided the need for passing `local bool` into Repository.Walk I'd be very happy. It's confusing to see method calls like 
```
err := source.Walk(filePath, false, func(prefix, name string) error {
```
because it's not obvious what 'false' means in that context. Also, 
```
repoBase = filepath.Join(r.LocalPath, "config")
```
Earlier in copy.go we saw, 
```
// pathForDestServiceConfig defines where in a 'gitops' repository the local config file
// for a given service should live.
func pathForDestServiceConfig(serviceName, name string) string {
	return filepath.Join("services/", serviceName, "base", name)
}
```
Even if you can refactor out a local Source.Walk() it would still be good to keep filepath manipulation in dedicated functions.

Reply:
OK.  I'll try it.